src/common/CurrencyEngine.ts

```diff
@@ -796,6 +796,14 @@ export class CurrencyEngine<
     return checkpointB
   }
 
+  updateTransactionEvents(): void {
+    if (this.transactionEvents.length > 0) {
+      this.walletLocalDataDirty = true
```

Comment:
The `CurrencyEngine.addTransaction` call already handles `walletLocalDataDirty = true` if there are any changes, so do we need it here?